`Development`: Add hint about the authorization object in the Java migration tool documentation (#4690)
ls1intum/Artemis: docs/dev/migration/data.rst

- All executed entries are saved in the table :code:`migration_changelog`. If you delete entries, they get executed again.
- Test your changes locally first, and only commit changes you are confident that work.
- Before deploying any database changes to a test server, ask for official permission from the project lead. If the changes don’t get approved, manual rollbacks can be necessary, which are avoidable.

> - Do not use repository methods that execute custom SQL/JPQL queries as this leads to problems.

Reviewer: ```suggestion
- If queries fail due to the authorization object being null, call `SecurityUtils.setAuthorizationObject()` beforehand to set a dummy object.
```

> - It is suggested to not use repository methods that execute custom SQL queries as this may lead to problems with the authentication to the database.

Reviewer: ```suggestion
- Do not use repository methods that execute custom SQL/JPQL queries as this leads to problems.
```

I think we can keep this short.
